gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/factories/GafferFactory.java

```diff
@@ -62,9 +64,8 @@
     public static Gaffer from(final GafferSpec gafferSpecConfig, final GaaSCreateRequestBody createGraphRequest) {
 
         // TODO: Validate only - and . special characters, see Kubernetes metadata regex
```

Comment:
Upon looking at it more, I think it is okay to validate in the api like you have done here:
 https://github.com/gchq/gaffer-experimental/blob/7f3973d0b959a79964036a411e9c54f6216d8906/gaffer-as-a-service/gaas-rest/src/main/java/uk/gov/gchq/gaffer/gaas/model/GaaSCreateRequestBody.java#L34

Perhaps though it would be nice if you could allow more characters (upper case or dashes) if possible to allow for multiple words